src/web/mjs/connectors/Komiku.mjs

```diff
@@ -9,23 +9,37 @@ export default class Komiku extends WordPressMangastream {
         super.label = 'Komiku';
         this.tags = ['manga', 'indonesian'];
         this.url = 'https://komiku.co.id';
-        this.path = '/daftar-komik/';
+        this.path = '/manga/page/';
 
-        this.queryMangas = 'div#a-z ol li.ranking1 h4';
+        this.queryMangas = 'div.daftar div.bge div.kan > a.popunder';
         this.queryChapters = 'table.chapter tbody tr td.judulseries a';
         this.queryChaptersTitle = undefined;
         this.queryPages = 'div#Baca_Komik img[src]:not([src=""])';
     }
 
     async _getMangas() {
-        let request = new Request(new URL(this.path, this.url), this.requestOptions);
-        let data = await this.fetchDOM(request, this.queryMangas);
-        return data.map(element => {
-            return {
-                id: this.getRootRelativeOrAbsoluteLink(element.closest('a'), request.url),
-                title: element.textContent.trim()
-            };
-        });
+        let mangaList = [];
+        for(let page = 1, run = true; run; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangas.length ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        try {
+            const uri = new URL(this.path + page + '/', this.url);
+            const request = new Request(uri, this.requestOptions);
+            const data = await this.fetchDOM(request, this.queryMangas);
+            return data.map(element => {
+                return {
+                    id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                    title: element.text.trim()
+                };
+            });
+        } catch(error) {
+            return [];
```

Comment:
This is probably to check if the last page was reached, but what happens when an error (e.g. 503) occurs right on page 1 or 2, the user will get an incomplete manga list without any indicator of an error.